pkg/http/sse.go

```diff
@@ -0,0 +1,97 @@
+package http
+
+import (
+	"net/http"
+
+	"github.com/ThreeDotsLabs/watermill/message"
+	"github.com/go-chi/render"
+)
+
+type StreamAdapter interface {
+	// ResponseProvider returns the response to be sent back to client.
+	// Any errors that occur should be handled and written to `w`, returning false as `ok`.
+	ResponseProvider(w http.ResponseWriter, r *http.Request) (response interface{}, ok bool)
+	// Validator validates if the incoming message should be handled by this handler.
+	// Typically this involves checking some kind of model ID.
+	Validator(r *http.Request, msg *message.Message) (ok bool)
+}
+
+type SSEHandler struct {
+	subscriber    message.Subscriber
+	topic         string
+	streamAdapter StreamAdapter
+}
+
+func NewSSEHandler(
+	subscriber message.Subscriber,
+	topic string,
+	streamAdapter StreamAdapter,
+) SSEHandler {
+	return SSEHandler{
+		subscriber:    subscriber,
```

Comment:
it should be very clear here, that every http need to have separated subscription which should be removed after request is done, maaaybe we can do it in some magic way, that one instance may get only one subscription which will send all messages to go-channel and every request will just subscribe to go-channel - but the question is how to handle it nice

Reply:
also, when we would like to use go-channel it's pretty important to have no leak etc. (so for example, if for some reason message here will be not consumed:
```
for msg := range messages {
			ok := s.streamAdapter.Validator(r, msg)
```
it should maybe ignore this message (as I know, they was some problems when client was stuck)

Reply:
So it's now taken care of by introducing the `SSERouter` component.